lib/greet.js

```diff
@@ -0,0 +1,9 @@
+'use strict';
+
+module.exports = exports = {};
+
+exports.helloFriend = function(name){
+  if (!name) throw new Error('Name not provided.');
```

Comment:
Read Feature Tasks carefully. If the argument passed is not a string, it should be returned as null. So line 6 should be changed to check if the arg is a string before moving forward.